File: Flutter/perimeterx/ios/Runner/PerimeterxManager.swift
import Foundation
import Flutter
import PerimeterX_SDK

class PerimeterxManager {
    
    static let shared = PerimeterxManager()
    
    func start() {
        do {
            let policy = PXPolicy()
            policy.urlRequestInterceptionType = .none
            policy.doctorCheckEnabled = true
            try PerimeterX.start(appId: "PXj9y4Q8Em", delegate: nil, policy: policy)
        }
        catch {
            print("error: \(error)")
        }
    }
    
    func setupChannel(with controller: FlutterViewController) {
        let perimeterxChannel = FlutterMethodChannel(name: "com.perimeterx.demo/perimeterx", binaryMessenger: controller.binaryMessenger)
        perimeterxChannel.setMethodCallHandler({(call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            if call.method == "_getPxHeaders" {
                var json: String?
                do {
                    if let headers = PerimeterX.headersForURLRequest() {
                        let data = try JSONSerialization.data(withJSONObject: headers)
                        json = String(data: data, encoding: .utf8)
                    }
                }
                catch {
                    print("error: \(error)")
                }
                result(json)
            }
            else if call.method == "_handlePxResponse" {
                if let response = call.arguments as? String, let data = response.data(using: .utf8), let httpURLResponse = HTTPURLResponse(url: URL(string: "https://fake.url.com")!, statusCode: 403, httpVersion: nil, headerFields: nil) {
                    let handled = PerimeterX.handleResponse(response: httpURLResponse, data: data) { challengeResult in
                        result(challengeResult == .solved ? "solved" : "cancelled")
                    }
                    if handled {
                        return
                    }
                }
                result("false")
            }
            else {
                result("")
            }
        })
    }
}

File: Flutter/HUMAN_Demo/android/app/src/main/kotlin/com/perimeterx/perimeterx/PerimeterxManager.kt
package com.perimeterx.perimeterx

import android.app.Application
import com.perimeterx.mobile_sdk.PerimeterX
import com.perimeterx.mobile_sdk.PerimeterXChallengeResult
import com.perimeterx.mobile_sdk.main.PXPolicy
import com.perimeterx.mobile_sdk.main.PXPolicyUrlRequestInterceptionType
import com.perimeterx.mobile_sdk.main.PXStorageMethod
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import org.json.JSONObject

class PerimeterxManager {

    companion object {

        fun start(application: Application) {
            try {
                val policy = PXPolicy()
                policy.storageMethod = PXStorageMethod.DATA_STORE
                policy.urlRequestInterceptionType = PXPolicyUrlRequestInterceptionType.NONE
                policy.doctorCheckEnabled = true
                PerimeterX.start(application, "PXj9y4Q8Em", null, policy)
            }
            catch (exception: Exception) {
                println("failed to start. exception: $exception")
            }
        }

        fun handleEvent(call: MethodCall, result: MethodChannel.Result) {
            if (call.method == "_getPxHeaders") {
                val json = JSONObject(PerimeterX.headersForURLRequest(null) as Map<*, *>?)
                result.success(json.toString())
            }
            else if (call.method == "_handlePxResponse") {
                val handled = PerimeterX.handleResponse(call.arguments!! as String, null) { challengeResult: PerimeterXChallengeResult ->
                    result.success(if (challengeResult == PerimeterXChallengeResult.SOLVED) "solved" else "cancelled")
                    null
                }
                if (!handled) {
                    result.success("false")
                }
            }
            else {
                result.notImplemented()
            }
        }
    }
}
